src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -17,6 +21,8 @@ internal class HostingDiagnosticListener : IApplicationInsightDiagnosticListener
         private readonly TelemetryClient client;
         private readonly ICorrelationIdLookupHelper correlationIdLookupHelper;
         private readonly string sdkVersion;
+        private bool? isAspNetCore20;
```

Comment:
could you make it static and initialize in place, so it don't have to be nullable, you can also remove property